src/core/corefile.cpp

```diff
@@ -55,13 +55,24 @@ unsigned CoreFile::corefileIterationInterval()
 void CoreFile::sendAvatarFile(Core* core, uint32_t friendId, const QByteArray& data)
 {
     QMutexLocker mlocker(&fileSendMutex);
-
-    static_assert(TOX_HASH_LENGTH <= TOX_FILE_ID_LENGTH, "TOX_HASH_LENGTH > TOX_FILE_ID_LENGTH!");
+    uint32_t fileNum;
     uint8_t avatarHash[TOX_HASH_LENGTH];
-    tox_hash(avatarHash, (uint8_t*)data.data(), data.size());
-    uint64_t filesize = data.size();
-    uint32_t fileNum = tox_file_send(core->tox, friendId, TOX_FILE_KIND_AVATAR, filesize,
-                                     avatarHash, avatarHash, TOX_HASH_LENGTH, nullptr);
+    uint64_t filesize;
+
+    if (!data.isEmpty())
+    {
+        static_assert(TOX_HASH_LENGTH <= TOX_FILE_ID_LENGTH, "TOX_HASH_LENGTH > TOX_FILE_ID_LENGTH!");
+        tox_hash(avatarHash, (uint8_t*)data.data(), data.size());
+        filesize = data.size();
+        fileNum = tox_file_send(core->tox, friendId, TOX_FILE_KIND_AVATAR, filesize,
+                                         avatarHash, avatarHash, TOX_HASH_LENGTH, nullptr);
+    }
+    else
+    {
+        fileNum = tox_file_send(core->tox, friendId, TOX_FILE_KIND_AVATAR, 0,
+                                         nullptr, nullptr, 0, nullptr);
+    }
```

Comment:
Nitpick, but if you turn just that else into:

```
if (data.isEmpty())
{
    fileNum = tox_file_send(core->tox, friendId, TOX_FILE_KIND_AVATAR, 0,
                                         nullptr, nullptr, 0, nullptr);
    return;
}
```

You can turn back the two `if (!data.isEmpty())` blocks at lines 62 and 83 into unconditional code.